diff --git a/build/comet38/comet38client.js b/build/comet38/comet38client.js
index 287e8da4fab1f5dc7a36015197a40a80f5c4da0d..7aa55ccebda0524af6de06fa895ff28bf9b92f17 100644
--- a/build/comet38/comet38client.js
+++ b/build/comet38/comet38client.js
@@ -47,7 +47,7 @@ class Comet38Client {
         // (our CI) when skipping the status call before doing other queries. Sleeping a little
         // while did not help. Thus we query the version as a way to say "hi" to the backend,
         // even in cases where we don't use the result.
-        const _version = await this.detectVersion(rpcClient);
+        // const _version = await this.detectVersion(rpcClient);
         return Comet38Client.create(rpcClient);
     }
     /**
diff --git a/build/tendermint34/tendermint34client.js b/build/tendermint34/tendermint34client.js
index f3f10bbc79aa203139730a19f4189a9d6a1cae97..c4431ed5fcdfbb0d088f36f488c6228bd44d14ac 100644
--- a/build/tendermint34/tendermint34client.js
+++ b/build/tendermint34/tendermint34client.js
@@ -47,7 +47,7 @@ class Tendermint34Client {
         // (our CI) when skipping the status call before doing other queries. Sleeping a little
         // while did not help. Thus we query the version as a way to say "hi" to the backend,
         // even in cases where we don't use the result.
-        const _version = await this.detectVersion(rpcClient);
+        // const _version = await this.detectVersion(rpcClient);
         return Tendermint34Client.create(rpcClient);
     }
     /**
diff --git a/build/tendermint37/tendermint37client.js b/build/tendermint37/tendermint37client.js
index 9e7153a66271fe960d1ff7608a9e4afe5ef10556..b0942c14016ddb193f08bf8c9c0b040f941d04e4 100644
--- a/build/tendermint37/tendermint37client.js
+++ b/build/tendermint37/tendermint37client.js
@@ -47,7 +47,7 @@ class Tendermint37Client {
         // (our CI) when skipping the status call before doing other queries. Sleeping a little
         // while did not help. Thus we query the version as a way to say "hi" to the backend,
         // even in cases where we don't use the result.
-        const _version = await this.detectVersion(rpcClient);
+        // const _version = await this.detectVersion(rpcClient);
         return Tendermint37Client.create(rpcClient);
     }
     /**
