diff --git a/build/signingcosmwasmclient.d.ts b/build/signingcosmwasmclient.d.ts
index 398df4c7b16e43f69b28d6a893ddfdd5061365f2..f0dc060c665522c47a9e1b833952df299ac1e0b4 100644
--- a/build/signingcosmwasmclient.d.ts
+++ b/build/signingcosmwasmclient.d.ts
@@ -143,7 +143,7 @@ export declare class SigningCosmWasmClient extends CosmWasmClient {
      */
     static offline(signer: OfflineSigner, options?: SigningCosmWasmClientOptions): Promise<SigningCosmWasmClient>;
     protected constructor(cometClient: CometClient | undefined, signer: OfflineSigner, options: SigningCosmWasmClientOptions);
-    simulate(signerAddress: string, messages: readonly EncodeObject[], memo: string | undefined): Promise<number>;
+    simulate(signerAddress: string, messages: readonly EncodeObject[], memo: string | undefined, explicitSignerData?: Partial<SignerData>): Promise<number>;
     /** Uploads code and returns a receipt, including the code ID */
     upload(senderAddress: string, wasmCode: Uint8Array, fee: StdFee | "auto" | number, memo?: string, instantiatePermission?: AccessConfig): Promise<UploadResult>;
     instantiate(senderAddress: string, codeId: number, msg: JsonObject, label: string, fee: StdFee | "auto" | number, options?: InstantiateOptions): Promise<InstantiateResult>;
@@ -169,7 +169,7 @@ export declare class SigningCosmWasmClient extends CosmWasmClient {
      * @param memo
      * @param timeoutHeight (optional) timeout height to prevent the tx from being committed past a certain height
      */
-    signAndBroadcast(signerAddress: string, messages: readonly EncodeObject[], fee: StdFee | "auto" | number, memo?: string, timeoutHeight?: bigint): Promise<DeliverTxResponse>;
+    signAndBroadcast(signerAddress: string, messages: readonly EncodeObject[], fee: StdFee | "auto" | number, memo?: string, timeoutHeight?: bigint, explicitSignerData?: SignerData): Promise<DeliverTxResponse>;
     /**
      * Creates a transaction with the given messages, fee, memo and timeout height. Then signs and broadcasts the transaction.
      *
@@ -185,7 +185,7 @@ export declare class SigningCosmWasmClient extends CosmWasmClient {
      *
      * @returns Returns the hash of the transaction
      */
-    signAndBroadcastSync(signerAddress: string, messages: readonly EncodeObject[], fee: StdFee | "auto" | number, memo?: string, timeoutHeight?: bigint): Promise<string>;
+    signAndBroadcastSync(signerAddress: string, messages: readonly EncodeObject[], fee: StdFee | "auto" | number, memo?: string, timeoutHeight?: bigint, explicitSignerData?: SignerData): Promise<string>;
     sign(signerAddress: string, messages: readonly EncodeObject[], fee: StdFee, memo: string, explicitSignerData?: SignerData, timeoutHeight?: bigint): Promise<TxRaw>;
     private signAmino;
     private signDirect;
diff --git a/build/signingcosmwasmclient.js b/build/signingcosmwasmclient.js
index 6eb95bb672698df8feba51dc5a06d33bbed616c5..58d2dd21d79370a4af726bcb607b8b72f7b598cc 100644
--- a/build/signingcosmwasmclient.js
+++ b/build/signingcosmwasmclient.js
@@ -93,7 +93,12 @@ class SigningCosmWasmClient extends cosmwasmclient_1.CosmWasmClient {
             throw new Error("Failed to retrieve account from signer");
         }
         const pubkey = (0, amino_1.encodeSecp256k1Pubkey)(accountFromSigner.pubkey);
-        const { sequence } = await this.getSequence(signerAddress);
+        let sequence;
+        if (explicitSignerData?.sequence !== undefined) {
+            sequence = explicitSignerData.sequence;
+        } else {
+            sequence = (await this.getSequence(signerAddress)).sequence;
+        }
         const { gasInfo } = await this.forceGetQueryClient().tx.simulate(anyMsgs, memo, pubkey, sequence);
         (0, utils_1.assertDefined)(gasInfo);
         return math_1.Uint53.fromString(gasInfo.gasUsed.toString()).toNumber();
@@ -327,18 +332,30 @@ class SigningCosmWasmClient extends cosmwasmclient_1.CosmWasmClient {
      * @param memo
      * @param timeoutHeight (optional) timeout height to prevent the tx from being committed past a certain height
      */
-    async signAndBroadcast(signerAddress, messages, fee, memo = "", timeoutHeight) {
+    async signAndBroadcast(signerAddress, messages, fee, memo = "", timeoutHeight, explicitSignerData) {
         let usedFee;
+        let signerData = explicitSignerData;
+        const { sequence, accountNumber } = explicitSignerData
+          ? explicitSignerData
+          : await this.getSequence(signerAddress);
         if (fee == "auto" || typeof fee === "number") {
             (0, utils_1.assertDefined)(this.gasPrice, "Gas price must be set in the client options when auto gas is used.");
-            const gasEstimation = await this.simulate(signerAddress, messages, memo);
+            const gasEstimation = await this.simulate(signerAddress, messages, memo, { sequence });
             const multiplier = typeof fee === "number" ? fee : this.defaultGasMultiplier;
             usedFee = (0, stargate_1.calculateFee)(Math.round(gasEstimation * multiplier), this.gasPrice);
         }
         else {
             usedFee = fee;
         }
-        const txRaw = await this.sign(signerAddress, messages, usedFee, memo, undefined, timeoutHeight);
+        if (!signerData) {
+            const chainId = await this.getChainId();
+            signerData = {
+              accountNumber: accountNumber,
+              sequence: sequence,
+              chainId: chainId,
+            };
+        }
+        const txRaw = await this.sign(signerAddress, messages, usedFee, memo, signerData, timeoutHeight);
         const txBytes = tx_3.TxRaw.encode(txRaw).finish();
         return this.broadcastTx(txBytes, this.broadcastTimeoutMs, this.broadcastPollIntervalMs);
     }
@@ -357,18 +374,31 @@ class SigningCosmWasmClient extends cosmwasmclient_1.CosmWasmClient {
      *
      * @returns Returns the hash of the transaction
      */
-    async signAndBroadcastSync(signerAddress, messages, fee, memo = "", timeoutHeight) {
+    async signAndBroadcastSync(signerAddress, messages, fee, memo = "", timeoutHeight, explicitSignerData) {
         let usedFee;
+        let signerData = explicitSignerData;
+        const { sequence, accountNumber } = explicitSignerData
+            ? explicitSignerData
+            : await this.getSequence(signerAddress);
         if (fee == "auto" || typeof fee === "number") {
             (0, utils_1.assertDefined)(this.gasPrice, "Gas price must be set in the client options when auto gas is used.");
-            const gasEstimation = await this.simulate(signerAddress, messages, memo);
+            const gasEstimation = await this.simulate(signerAddress, messages, memo, { sequence });
             const multiplier = typeof fee === "number" ? fee : this.defaultGasMultiplier;
             usedFee = (0, stargate_1.calculateFee)(Math.round(gasEstimation * multiplier), this.gasPrice);
         }
         else {
             usedFee = fee;
         }
-        const txRaw = await this.sign(signerAddress, messages, usedFee, memo, undefined, timeoutHeight);
+        if (!signerData) {
+            const chainId = await this.getChainId();
+            signerData = {
+              accountNumber: accountNumber,
+              sequence: sequence,
+              chainId: chainId,
+            };
+        }
+      
+        const txRaw = await this.sign(signerAddress, messages, usedFee, memo, signerData, timeoutHeight);
         const txBytes = tx_3.TxRaw.encode(txRaw).finish();
         return this.broadcastTxSync(txBytes);
     }
